includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -1603,22 +1603,25 @@ public function link_session_to_customer( $session_id, $customer_id ) {
 		);
 	}
 
+
 	/**
-	 * Registers a new domain with Apple Pay.
+	 * Registers a Payment Method Domain.
 	 *
-	 * @param string $domain_name Domain name which to register for Apple Pay.
+	 * @param string $domain_name Domain name which to register for the account.
 	 *
-	 * @return array An array containing an id in case it has succeeded, or an error message in case it has failed.
+	 * @return array An array containing an id and the bool property 'enabled' indicating
+	 * whether the domain is enabled for the account. Each Payment Method
+	 * (apple_pay, google_pay, link, paypal) in the array have a 'status'
+	 * property with the possible values 'active' and 'inactive'.
 	 *
 	 * @throws API_Exception If an error occurs.
 	 */
-	public function register_domain_with_apple( $domain_name ) {
+	public function register_domain( $domain_name ) {
 		return $this->request(
 			[
-				'test_mode'   => false, // Force live mode - Domain registration doesn't work in test mode.
 				'domain_name' => $domain_name,
 			],
```

Comment:
I think we should also explicitly [enable the domain](https://docs.stripe.com/api/payment_method_domains/create#create_payment_method_domain-enabled) by sending `'enabled' => true,` here. Enabling the GPay/APay buttons in WooPayments is explicitly trying to enable the buttons, and thus not only is the domain registration required, but also actually enabling the domain.

```php
			[
				'domain_name' => $domain_name,
				// ⛔️ For some reason this needs to be a string. If it's an actual boolean it's serialized as a 1 (integer) when forwarded by the server causing an invalid request error being returned from Stripe.
				'enabled' => 'true',
			],
```

With a corresponding change to allow the `enabled` parameter on the server.

You can test that this is working as intended by:

```sh
# Find the payment method domain ID you want to disable from the list of available domains.
# Looks like `pmd_abc123`.
curl https://api.stripe.com/v1/payment_method_domains \
  -u "<secret_key>:" \
  -H "Stripe-Account: <acct_id>"

# Disable the domain.
curl https://api.stripe.com/v1/payment_method_domains/<pmd_abc123> \
  -u "<secret_key>:" \
  -H "Stripe-Account: <acct_id>" \
  -d enabled=false

```

Then go disable Apple Pay/Google Pay, and re-enable them.

Reply:
Thank you for the suggestion. Added on https://github.com/Automattic/woocommerce-payments/pull/9139/commits/8a5f32ef9f2d96a5fa1ba337479eb2ed019166ae